packages/server/src/arpa_reporter/routes/exports.js

```diff
@@ -14,8 +53,25 @@ router.get('/', requireUser, async (req, res) => {
         res.status(404).json({ error: 'invalid reporting period' });
         return;
     }
+    const tenantId = useTenantId();
+
+    if (req.query.async) {
+        // Special handling for async treasury report generation and sending.
+        console.log('/api/exports?async=true GET');
+        console.log('Generating Async treasury report');
+        try {
+            const user = useUser();
+            arpa.generateAndSendEmail(user.email, periodId, tenantId);
```

Comment:
We should `await` here, otherwise it could fail after the success.

Reply:
Here (and similarly for the audit-report) I made a deliberate trade-off call. Lmk if we can handle this more gracefully.

The constraints are:
- The aws api-gateway request timeout is 30 seconds
- Report generation for partners with larger files can be in order of minutes

I felt it made sense to tell the client asap that their request was successful here and find an alternate means to notify the client of a failure if it occurred a few minutes into the file processing. Currently there is no "failure email" setup but that seemed like a good next-step.

wdyt?

Reply:
That makes sense. I don't think there is an easier way to handle this more gracefully. We can discuss on Friday, but it would require more work and thought around the proper design. This looks good for now. Thanks for the explanation.